src/openfermion/utils/_trotter_exp_to_qgates.py

```diff
@@ -0,0 +1,329 @@
+        term_coeff = qubit_operator.terms[term]
+
+        # List of operators and list of qubit ids
+        ops = []
+        qids = []
+        string_basis_1 = []  # Basis rotations 1
+        string_basis_2 = []  # Basis rotations 2
+
+        for p in term:  # p = single pauli term
+            qid = p[0]  # Qubit index
+            pop = p[1]  # Pauli op
+
+            qids.append(qid)  # Qubit index
+            ops.append(pop)  # Pauli operator
+
+            if pop == 'X':
+                string_basis_1.append("H {}".format(qid))    # Hadamard
+                string_basis_2.append("H {}".format(qid))    # Hadamard
+            elif pop == 'Y':
+                string_basis_1.append("Rx 1.57079632679 {}".format(qid))
+                string_basis_2.append("Rx -1.57079632679 {}".format(qid))
+
+        # Prep for CNOTs
+        cnot_pairs = numpy.vstack((qids[:-1], qids[1:]))
+        cnots1 = []
+        cnots2 = []
+        for i in range(cnot_pairs.shape[1]):
+            pair = cnot_pairs[:, i]
+            cnots1.append("CNOT {} {}".format(pair[0], pair[1]))
+        for i in numpy.arange(cnot_pairs.shape[1])[::-1]:
+            pair = cnot_pairs[:, i]
+            cnots2.append("CNOT {} {}".format(pair[0], pair[1]))
+
+        # Exponentiating each Pauli string requires five parts
+
+        # 1. Perform basis rotations
+        output_stream.write('\n'.join(string_basis_1))
+        if not len(string_basis_1) == 0:
+            output_stream.write('\n')
+
+        # 2. First set CNOTs
+        output_stream.write('\n'.join(cnots1))
+        output_stream.write('\n')
+
+        # 3. Rotation (Note kexp & Ntrot)
+        output_stream.write("Rz {} {}\n".format(term_coeff, qids[-1]))
+
+        # 4. Second set of CNOTs
+        output_stream.write('\n'.join(cnots2))
+        output_stream.write('\n')
+
+        # 5. Rotate back to Z basis
+        output_stream.write('\n'.join(string_basis_2))
+        if not len(string_basis_2) == 0:
+            output_stream.write('\n')
+
+
+def product_exp_pauli_strings_to_qasm(output_stream, op_list):
+    """Sends a list of QubitOperators to be exponentiated one by one
+    and printed to a QASM stream one at a time. The input is a list of
+    Pauli-strings and the output is the ordered product of their
+    exponents.
+
+    Args:
+    output_stream (file/string/IO stream): QASM output stream
+    op_list(list of QubitOperators): list of QubitOperators to be
+        exponentiated
+
+    Returns:
+        None
+    """
+
+    for op in op_list:
+        exp_sgl_pauli_string_to_qasm(output_stream, op)
+
+
+def trotterize_exp_qubop_to_qasm(
+                output_stream,
+                hamiltonian,
+                trotter_number=None,
+                trotter_order=None,
+                term_ordering=None,
+                k_exp=None):
+    """Trotterize a Qubit hamiltonian, exponentiate each trotterized term
+    and write it to a QASM output stream.
```

Comment:
Keep these parts to 1 line, but add more description below.  Say exactly what is being computed if possible.  For example, are you assuming the hamiltonian/qubit operator has already had a -1.0j or 1.0j applied? I.e. is it Hermitian or anti-Hermitian upon input? What's the sign convention?

Reply:
Where should this description go? Still in the docstrings?

Reply:
Yes in the doc, but not in the topline.  It's standard convention to have a 1 line short description, a space, then a longer description that gives more detail about the function.